core/src/test/scala/FSKafkaAlgebraSpec.scala

```diff
@@ -18,19 +18,33 @@ package freestyle
 package kafka
 
 import cats.MonadError
-import cats.instances.either._
+import cats.implicits._
 import freestyle.async.{AsyncContext, Proc}
 import net.manub.embeddedkafka.EmbeddedKafka
+import org.apache.kafka.clients.consumer.KafkaConsumer
 import org.apache.kafka.clients.producer.KafkaProducer
-import org.apache.kafka.common.serialization.Serializer
+import org.apache.kafka.common.serialization.{Deserializer, Serializer}
 import org.scalatest._
 import org.scalatest.concurrent.{ScalaFutures, Waiters}
+import freestyle._
+import freestyle.implicits._
 
-trait FSKafkaAlgebraSpec extends EmbeddedKafka with Waiters with Matchers with Implicits {
-  self: Suite =>
+trait FSKafkaAlgebraSpec
+    extends EmbeddedKafka
+    with Waiters
+    with Matchers
+    with Implicits
+    with BeforeAndAfterAll { self: Suite =>
 
   type Target[A] = Either[Throwable, A]
 
+  abstract override def run(testName: Option[String], args: Args) = {
+    EmbeddedKafka.start()
```

Comment:
Out of curiosity, why not put `EmbeddedKafka.start()` and `EmbeddedKafka.stop()` inside `beforeAll` and `afterAll`, respectively, and not override `run`. All three are inherited from `BeforeAndAfterAll`.